packages/test/test-end-to-end-tests/src/test/dataStoresNested.spec.ts

```diff
@@ -0,0 +1,194 @@
+		registryEntries: [[dataStoreFactory.type, Promise.resolve(dataStoreFactory)]],
+		runtimeOptions,
+	});
+
+	async function addContainer(container: IContainer) {
+		containers.push(container);
+		const dataStores = (await container.getEntryPoint()) as IDataStores;
+
+		await provider.ensureSynchronized();
+
+		return { container, dataStores };
+	}
+
+	const createContainer = async () => {
+		const container = await provider.createContainer(runtimeFactory);
+		return addContainer(container);
+	};
+
+	async function addContainerInstance() {
+		const container = await provider.loadContainer(runtimeFactory);
+		return addContainer(container);
+	}
+
+	beforeEach("getTestObjectProvider", async () => {
+		const driver = new LocalServerTestDriver();
+		const registry = [];
+		seed = 1; // Every test is independent from another test!
+
+		provider = new TestObjectProvider(
+			Loader,
+			driver,
+			() =>
+				new TestContainerRuntimeFactory(
+					"@fluid-experimental/test-dataStores",
+					new TestFluidObjectFactory(registry),
+				),
+		);
+		// syncSummarizer: true
+		provider.resetLoaderContainerTracker(true); // syncSummarizerClients
+
+		loader = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+	});
+
+	afterEach(() => {
+		provider.reset();
+		for (const container of containers) {
+			container.close();
+		}
+		containers = [];
+		summaryCollection = undefined;
+		summarizer = undefined;
+		loader = undefined;
+	});
+
+	async function waitForSummary(): Promise<string> {
+		// ensure that all changes made it through
+		await provider.ensureSynchronized();
+
+		assert(summaryCollection !== undefined, "summary setup properly");
+		// create promise before we call summarizeNow, as otherwise we might miss summary and will wait
+		// forever for next one to happen
+		const wait = summaryCollection.waitSummaryAck(
+			containers[0].deltaManager.lastSequenceNumber,
+		);
+		const summaryResult = await summarizeNow(summarizer!);
+		assert(summaryResult.summaryVersion !== undefined, "summary result");
+		const ackedSummary = await wait;
+		assert(ackedSummary.summaryAck.contents.handle !== undefined, "summary acked");
+		return summaryResult.summaryVersion;
+	}
+
+	/**
+	 * Creates a pair of containers and initializes them with initial state
+	 * @returns collab space
+	 */
+	async function initialize() {
+		const { container, dataStores } = await createContainer();
+
+		// Create and setup a summary collection that will be used to track and wait for summaries.
+		summaryCollection = new SummaryCollection(container.deltaManager, createChildLogger());
+
+		const summarizerRes = await createSummarizerCore(container, loader!);
+		summarizer = summarizerRes.summarizer;
+		// containers.push(summarizerRes.container);
+
+		// Have a second container that follows passivley the first one
+		await addContainerInstance();
+
+		await provider.ensureSynchronized();
+
+		return dataStores;
+	}
+
+	it("Basic test", async () => {
```

Comment:
#TODO add tests including aliasing

Reply:
Hm, I do not want to expose or use aliasing on second level. Maybe we want to add it for completeness (such that we can validate it works on second level and would not be broken in the future), but that's like Pri4 for me.

Reply:
@vladsud so aliasing works from IDataStore. Because of this, we only need to worry about it based on what we decide the final DataStores API would look like.

As you will be the first user of this API, you get to call the shots here. In this PR, it's using `_createFluidDataStoreContext` which returns a `IFluidDataStoreChannel` and that's it. (**Approach A**)

Another approach would be to unify the APIs and have the DataStores have a similar API as the runtime and one can create datastores which are of type `IDataStore` with the caveat that for now, `trySetAlias` will fail with a `UsageError` for nested datastores (trySetAlias will for now only work for data stores created by the runtime). (**Approach B**).

Another (nested) approach (no pun intended) is to have `trySetAlias` work for nested datastores as well, but at the runtime level in terms of namespace collision. I don't like this approach because it does not play well with the long-term goal of having alias working properly at the level it's being called on (hopefully, at least that's how I see things). (**Approach B'**)

I would go with **Approach B**. What do you think, Vlad?

Reply:
Approach A is really bad - we kept saying that context & channel interfaces are only for child-parent relationship and should not be used anywhere else. We violate that rule with detached context creation (do not confuse with detached data store), but overall, that holds water.

It has to be some version of Approach B. While I do not need it now, eventually I think we will need to have similar to ContainerRuntime API - if anything, Loop would need to pass around notion of "parent" around that is a home for sub-components. It does not have to be full IContainerRuntimeBase, but it will need to some sub-set. Such API could be directly provided by DataStores, or it could be responsibility of who uses DataStores (me) and exposes such API, but the closer they are, the easier it is.

I think aliasing should work. In a sense that ops would flow and alias would be assigned correctly (with a scope of nested data store). As long as we do not load from summary :)
Summaries will not work due to same problem as everywhere else with this prototype - responsibilities are not properly divided across layers. We should revisit addContainerStateToSummary() and find a way for DataStore instance to record aliasBlobName blob. As of now, nested DataStore will never record it in a summary

The alternative design - aliasing is global, and nested data stores use "A/B" aliases. I do not like this direction.